transport/http_util.go

```diff
@@ -463,14 +465,16 @@ func encodeGrpcMessage(msg string) string {
 
 func encodeGrpcMessageUnchecked(msg string) string {
 	var buf bytes.Buffer
-	lenMsg := len(msg)
-	for i := 0; i < lenMsg; i++ {
-		c := msg[i]
-		if c >= spaceByte && c < tildaByte && c != percentByte {
-			buf.WriteByte(c)
-		} else {
-			buf.WriteString(fmt.Sprintf("%%%02X", c))
+	for len(msg) > 0 {
+		r, size := utf8.DecodeRuneInString(msg)
+		for _, b := range []byte(string(r)) {
+			if r >= spaceByte && r < tildaByte && r != percentByte {
```

Comment:
How about adding `if size == 1 && ...`, since in this way, it will save 1 comparison for each byte when the utf-8 encoding takes more than 1 byte.

Reply:
Done!